pkg/secrets/secrets.go

```diff
@@ -41,6 +42,8 @@ const (
 // SecretDefinition is used for representing
 // a secret definition input
 type SecretDefinition struct {
+	sync.RWMutex
```

Comment:
It looks like the `RLock()` calls are missing for when the `secrets` is read from after secrets are fetched, is that alright because we essentially have a "write barrier" for fetching all secrets in parallel before they are read out?  Should this just be a normal sync.Mutex instead?

Reply:
> is that alright because we essentially have a "write barrier" for fetching all secrets in parallel before they are read out?

In the current design, yes. I used RWMutex with a bit of future proofing in mind; right now outputting the secrets is a serial operation that occurs after the parallel-read. I could see a future where each secret definition's read AND write can happen in parallel in order to produce runtime. I am open to switching it to `sync.Mutex` to keep it simple.

Reply:
Ah ok, that makes sense.  We'll have to keep that in mind if we decide to change the design as you mentioned.